### PR TITLE
added health card to object service dashboard

### PR DESCRIPTION
**Pull request by afreen23**
mine

### CODE REVIEW

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import { HealthState } from '@console/internal/components/dashboard/health-card/states';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { NooBaaSystemModel } from '../../models';
+
+enum HealthCardQueries {
+  BUCKETS_COUNT = 'NooBaa_num_buckets',
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: can you move these queries to `queries.ts` and also move all existing ones too (from Buckets card) ? See  https://github.com/openshift/console/pull/1732/files#diff-0d0741fd0d5bcb5571ab366134c35eebR1

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  bucketsResponse,
+  unhealthyBucketsResponse,
+  poolsResponse,
+  unhealthyPoolResponse,
+  noobaaSystemData,
+): ObjectStorageHealth => {
+  if (!(bucketsResponse && unhealthyBucketsResponse && poolsResponse && unhealthyPoolResponse)) {
+    return { state: HealthState.LOADING };
+  }
+  const buckets = _.get(bucketsResponse, 'data.result[0].value[1]');
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: Buckets card already has this util function https://github.com/openshift/console/blob/master/frontend/packages/noobaa-storage-plugin/src/components/buckets-card/buckets-card.tsx#L26 which should be reused here too. Maybe move the function to some util file.

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+};
+
+const getObjectStorageHealthState = (
+  bucketsResponse,
+  unhealthyBucketsResponse,
+  poolsResponse,
+  unhealthyPoolResponse,
+  noobaaSystemData,
+): ObjectStorageHealth => {
+  if (!(bucketsResponse && unhealthyBucketsResponse && poolsResponse && unhealthyPoolResponse)) {
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: what about loaded state of `noobaaSystemData` ? Shouldn't that be considered too ?

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { HealthBody, HealthItem } from '@console/internal/components/dashboard/health-card';
+import { HealthState } from '@console/internal/components/dashboard/health-card/states';
+import {
+  withDashboardResources,
+  DashboardItemProps,
+} from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: please add empty line between `@console/internal/module/k8s` and `../../models` imports

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { HealthBody, HealthItem } from '@console/internal/components/dashboard/health-card';
+import { HealthState } from '@console/internal/components/dashboard/health-card/states';
+import {
+  withDashboardResources,
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: alpha

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx (hunk 4 above)
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: alpha

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx (hunk 3 above)
Comment (afreen23) [reply frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: yes I missed it !

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+    HealthCardQueries.UNHEALTHY_BUCKETS,
+    'result',
+  ]);
+  const poolsQueryResult = prometheusResults.getIn([HealthCardQueries.POOLS_COUNT, 'result']);
+  const unhealthyPoolsQueryResult = prometheusResults.getIn([
+    HealthCardQueries.UNHEALTHY_POOLS,
+    'result',
+  ]);
+
+  const noobaaSystem = _.get(resources, 'noobaa');
```
Comment (rawagner) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]:
```suggestion
  const noobaaSystemData = _.get(resources.noobaa, 'data') as K8sResourceKind;
```

File: frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+} from '@console/internal/components/dashboard/dashboard-card';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { HealthBody, HealthItem } from '@console/internal/components/dashboard/health-card';
+import { HealthState } from '@console/internal/components/dashboard/health-card/states';
+import {
```
Comment (cloudbehl) [inline frontend/packages/noobaa-storage-plugin/src/components/health-card/health-card.tsx]: alpha

**Pull request merged**